11_deploy_kafka.sh

```diff
@@ -0,0 +1,75 @@
+oc apply -f install/cluster-operator/031-RoleBinding-strimzi-cluster-operator-entity-operator-delegation.yaml -n ${KAFKA_NAMESPACE}
+oc apply -f install/cluster-operator/032-RoleBinding-strimzi-cluster-operator-topic-operator-delegation.yaml -n ${KAFKA_NAMESPACE}
+
+# Install Operator
+oc apply -f install/cluster-operator -n ${KAFKA_NAMESPACE}
+sleep 5
+oc wait --for condition=ready pod -l name=strimzi-cluster-operator -n ${KAFKA_NAMESPACE} --timeout=120s
+
+# Modify Kafka cluster & Deploy
+sed -i "s/my-cluster/${KAFKA_CLUSTERNAME}/" metrics/examples/kafka/kafka-metrics.yaml
+sed -i "s/100Gi/${KAFKA_PVC_SIZE}Gi/" metrics/examples/kafka/kafka-metrics.yaml
+sed -i "s/my-cluster/${KAFKA_CLUSTERNAME}/" metrics/examples/kafka/kafka-connect-metrics.yaml
+sed -i "s/my-connect/${KAFKA_CLUSTERNAME}/" metrics/examples/kafka/kafka-connect-metrics.yaml
+oc apply -f metrics/examples/kafka/kafka-metrics.yaml -n ${KAFKA_NAMESPACE}
+sleep 5
+oc wait --for condition=ready pod -l strimzi.io/cluster=${KAFKA_CLUSTERNAME} -n ${KAFKA_NAMESPACE} --timeout=120s
+oc apply -f metrics/examples/kafka/kafka-connect-metrics.yaml -n ${KAFKA_NAMESPACE}
+sleep 5
+oc wait --for condition=ready pod -l strimzi.io/kind=KafkaConnect -n ${KAFKA_NAMESPACE} --timeout=240s
+
+# Modify Prometheus & Deploy
+sed -i "s/myproject/${KAFKA_CLUSTERNAME}/" metrics/examples/prometheus/prometheus.yaml
+oc apply -f metrics/examples/prometheus/prometheus.yaml -n ${KAFKA_NAMESPACE}
+oc apply -f metrics/examples/prometheus/alerting-rules.yaml -n ${KAFKA_NAMESPACE}
+oc wait --for condition=ready pod -l name=prometheus -n ${KAFKA_NAMESPACE} --timeout=120s
+oc apply -f metrics/examples/prometheus/alertmanager.yaml -n ${KAFKA_NAMESPACE}
+oc wait --for condition=ready pod -l name=alertmanager -n ${KAFKA_NAMESPACE} --timeout=120s
+
+# Deploy Grafana
+oc apply -f metrics/examples/grafana/grafana.yaml -n ${KAFKA_NAMESPACE}
+oc wait --for condition=ready pod -l name=grafana -n ${KAFKA_NAMESPACE} --timeout=120s
+
+# Expose Grafana & Prometheus
+oc expose svc prometheus -n ${KAFKA_NAMESPACE} || echo "Prometheus route already exists" 
+oc expose svc grafana -n ${KAFKA_NAMESPACE} || echo "Grafana route already exists" 
+
+# Recover Grafana Dashboard
+wget -q https://raw.githubusercontent.com/ppatierno/rh-osd-2018/master/grafana-dashboards/strimzi-kafka.json -O metrics/examples/grafana/strimzi-kafka.json
+wget -q https://raw.githubusercontent.com/ppatierno/rh-osd-2018/master/grafana-dashboards/strimzi-zookeeper.json -O metrics/examples/grafana/strimzi-zookeeper.json
+
+# Add Grafana Dashboards and Datasource
+GRAFANA_ROUTE=`oc get route grafana --template='{{ .spec.host }}'`
+PROMETHEUS_ROUTE=`oc get route prometheus --template='{{ .spec.host }}'`
+curl -X "POST" "http://${GRAFANA_ROUTE}/api/datasources" -H "Content-Type: application/json" --user admin:admin --data-binary '{ "name":"Prometheus","type":"prometheus","access":"proxy","url":"http://'${PROMETHEUS_ROUTE}'","basicAuth":false,"isDefault":true }'
```

Comment:
To avoid traffic going out of the cluster, then going back, wouldn't be better to use the prometheus service instead?

Reply:
I also think that is the best way, but I've been suffering some fails regarding DNS issues and for now I think it's better to just use the external. I will fill an issue to change it to the SVC when it stays stable. Makes sense?